src/lib/server/twilio.ts

```diff
@@ -229,3 +241,53 @@ export const getMsg = async (url: URL) => {
 
 	return response;
 };
+
+export async function sendNotif() {
+	const nowLocal = new Date();
+	const users = await prisma.user.findMany({
+		select: {
+			id: true,
+			phone: true,
+			reminderDatetime: true,
+			reminderIntervalDays: true,
+			timeZone: true,
+			phonePermissions: {
+				select: {
+					allowReminders: true,
+					blocked: true
+				}
+			}
+		}
+	});
+	users.forEach(async (user) => {
+		const { id, phone, reminderDatetime, reminderIntervalDays, phonePermissions, timeZone } = user;
+		const { allowReminders, blocked } = phonePermissions;
+		if (!allowReminders || blocked) return;
+
+		const options = {
+			timeZone
+		};
+
+		const formattedDate = nowLocal.toLocaleString('en-US', options);
+		const now = new Date(formattedDate);
+		const timeDifference = Math.abs(now.getTime() - reminderDatetime.getTime()); // Get the absolute time difference in milliseconds
+		const minuteInMillis = 60 * 1000; // 1 minute in milliseconds
+		if (timeDifference < minuteInMillis) {
+			// currently within a minute of when user should be reminded
+			// send notif
+			await sendMsg({ phone, type: 'reminder' }, null);
+
+			// update reminder date for next notif
+			const newReminderDate = new Date(reminderDatetime);
+			newReminderDate.setDate(reminderDatetime.getDate() + reminderIntervalDays);
+			await prisma.user.update({
```

Comment:
There could also potentially be a race condition if tasks overlap. In fact, if for any reason the cron job is scheduled twice, then users will almost certainly get duplicate messages.

To make this very unlikely, you can do the following:

1. randomize the order of 'users' 
2. sleep for a random amount of time between 1 and 2 seconds prior to each message.
3. Re-query the database for the reminderDateTime before the time comparison logic. Using the cached data from the initial query greatly increases the odds of duplicate messages.

This will make it both robust in the face of unexpected concurrency as well as help avoid rate limiting issues.

Reply:
mmm sounds good

Reply:
will try to get this in tmrw~

Reply:
pushed~